piff/input.py

```diff
@@ -572,7 +613,7 @@ def readStarCatalog(self, cat_file_name, cat_hdu, x_col, y_col, flag_col, use_co
             cat = cat[cat[use_col]!=0]
 
         # Limit to nstars objects
-        if nstars is not None and nstars > len(cat):
+        if nstars is not None and nstars < len(cat):
             logger.info("Limiting to %d stars for %s",nstars,cat_file_name)
             cat = cat[:nstars]
```

Comment:
I only use the nstars feature to save time when debugging stuff.  If I'm just checking whether the code runs through to the end and does something reasonable, a lot of times it's handy to limit the number of stars being processed to only 20 or so per chip.  In which case, I don't care much which 20 it is.

I think probably if people want to select a subset of the full catalog where they would care that is not just the first N objects, they can probably add either a flag or use column and specify which ones to use that way.  We could also add a generic boolean `use` field that could be any calculation the user specifies.  But maybe I'll wait until someone actually has a concrete use case to help guide the API.